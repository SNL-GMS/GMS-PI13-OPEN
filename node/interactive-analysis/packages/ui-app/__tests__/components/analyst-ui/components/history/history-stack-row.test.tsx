import Enzyme from 'enzyme';
import React from 'react';
import {
  HistoryStackRow,
  HistoryStackRowProps
} from '../../../../../src/ts/components/analyst-ui/components/history/history-stack/history-stack-row';

describe('history stack row', () => {
  const history: HistoryStackRowProps = {
    isFirstRow: true,
    areUndoRedoAdjacent: false,
    redoTarget: true,
    undoTarget: false,
    areUndoRedoJoined: true
  };
  const child = <div />;
  test('renders correctly', () => {
    const wrapper = Enzyme.shallow(<HistoryStackRow {...history}>{child}</HistoryStackRow>);
    expect(wrapper).toMatchSnapshot();
  });
});
