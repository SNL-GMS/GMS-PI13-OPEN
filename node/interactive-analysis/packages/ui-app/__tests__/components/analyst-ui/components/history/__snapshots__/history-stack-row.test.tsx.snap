// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`history stack row renders correctly 1`] = `
<li
  className="list__row entry-row
      
      action-indicator-top "
>
  <div
    className="list__column list__column--meta-container
        bottom
        
        mixed"
  >
    <span
      className="action-indicator__text"
    />
    <span
      className="action-indicator__text"
    >
      REDO
    </span>
  </div>
  <div />
</li>
`;
